apis/python/src/tiledbsoma/_collection.py

```diff
@@ -572,7 +570,7 @@ def _my_repr(self) -> str:
     def _contents_lines(self, last_indent: str) -> Iterable[str]:
         indent = last_indent + "    "
         if self.closed:
-            return ()
```

Comment:
Is this doing what we expect it to be doing?

Reply:
I believe so - we simply want the iterable to end, which it does.  No idea why the code used to return an empty tuple, which is definitely wrong for an iterator of strings

Reply:
I think I was the author here, and it turns out this was a misconception that I had. What I didn’t realize was that the mere presence of a `yield` in a function turned it into a generator; it didn’t _just_ happen when the `yield` statement was hit.

So in my head:

```python
def some_fn(*, do_iter: bool) -> object:
  if do_iter:
    # If you hit this line, the function suspends itself and returns
    # a generator which yields the string "5" and then is exhausted.
    yield "5"
  else:
    # This is the part where I was WRONG: I thought that
    # if you hit this line, the function simply returns the value 1.
    return 1

some_fn(do_iter=False)
# -> 1
# i.e., list(some_fn(do_iter=False)) -> list(1) -> TypeError

some_fn(do_iter=True)
# -> generator which yields "5", then exits.
# i.e., list(some_fn(do_iter=True)) -> ["5"]
```

This is not the case; instead what happens is that the fact that there is a `yield` statement in the function at all means it is _always_ a generator. The value after the `return` is ignored, so what actually happens is:

```python
some_fn(do_iter=False)
# -> generator which immediately exits
# i.e., list(some_fn(do_iter=False)) -> []
```

For the case where I was “returning” `()`, that meant that we were actually returning an empty iterator, which was what I wanted to happen anyway. I’m surprised no tools actually caught that there shouldn’t be a value there!